src/inmemory_cache.rs

```diff
@@ -139,18 +176,14 @@ impl InMemoryCache for FoyerCache {
 pub(crate) fn create_block_cache(
     options: Option<InMemoryCacheOptions>,
 ) -> Option<Arc<dyn InMemoryCache>> {
-    if let Some(options) = options {
-        match options.cache_type {
-            CacheType::Moka => Some(Arc::new(MokaCache::new(options))),
-            CacheType::Foyer => Some(Arc::new(FoyerCache::new(options))),
-        }
-    } else {
-        None
-    }
+    options.map(|options| match options.cache_type {
+        CacheType::Moka => Arc::new(MokaCache::new(options)) as Arc<dyn InMemoryCache>,
+        CacheType::Foyer => Arc::new(FoyerCache::new(options)) as Arc<dyn InMemoryCache>,
+    })
 }
 
 /// wrapper around Option<CachedBlock> to provide helper functions
-pub(crate) enum CachedBlockOption {
+pub enum CachedBlockOption {
```

Comment:
I think this needs to get replaced with a trait and per-impl implementations (MokaCachedBlockOption, FoyerCachedBlockOption).

Reply:
I've created a new trait called `CachedEntry` and implemented it for both caches.